internal/atomicutil/value_test.go

```diff
@@ -19,3 +20,39 @@ func TestValue(t *testing.T) {
 		assert.Equal(t, 0, v.Load())
 	})
 }
+
+func TestStore(t *testing.T) {
+	v := NewValue(5)
+	v.Store(42)
+	assert.Equal(t, 42, v.Load())
+}
+
+func TestSwap(t *testing.T) {
+	v := NewValue(42)
+	v.Swap(33)
+	assert.Equal(t, 33, v.Load())
+}
+
+func TestCompareAndSwap(t *testing.T) {
+	v := NewValue(42)
+
+	swapped := v.CompareAndSwap(42, 33)
+	assert.Equal(t, true, swapped)
+	assert.Equal(t, 33, v.Load())
+
+	swapped = v.CompareAndSwap(42, 33)
+	assert.Equal(t, false, swapped)
+	assert.Equal(t, 33, v.Load())
+}
+
+// This is just to illustrate how we should use atomic.Pointer[T] instead.
```

Comment:
I agree it would be good to standardize on atomic.Pointer wherever it makes sense, but I think it might be more helpful to explain the difference(s) between them rather than just saying "use atomic.Pointer[T] instead."

Would this be an accurate summary of this test case?
```suggestion
// Unlike atomic.Pointer[T], calling Load() on an uninitialized Value[T] will 
// return the zero value for type T.
```

Reply:
OK yes, sounds good. I've filed an issue internally to discuss migrating to atomic.Pointer throughout.